### PR TITLE
main chapter

### PR DESCRIPTION
**Pull request by VolodymyrKirichenko**

### CODE REVIEW

File: client/src/components/Filters/Inputs/SortDirectionField/SortDirectionField.tsx [hunk 1]
```diff
@@ -12,7 +14,9 @@ export const SortDirectionField = () => {
       name="sortDirection"
       render={({ input }) => (
         <FormControl>
-          <FormLabel id="sort_direction">sort_direction</FormLabel>
+          <FormLabel id="sort_direction">
+            <FormattedMessage id="sort_direction.sort" />
+          </FormLabel>
           <RadioGroup
```
Comment (SerhiiKirik) [inline client/src/components/Filters/Inputs/SortDirectionField/SortDirectionField.tsx:20]:
```suggestion
          </FormLabel>
          
          <RadioGroup
```

File: client/src/components/Filters/Inputs/SortDirectionField/SortDirectionField.tsx [hunk 2]
```diff
@@ -4,6 +4,8 @@ import RadioGroup from '@mui/material/RadioGroup';
 import FormControlLabel from '@mui/material/FormControlLabel';
 import FormControl from '@mui/material/FormControl';
 import FormLabel from '@mui/material/FormLabel';
+import { FormattedMessage } from 'react-intl';
+import React from 'react';
```
Comment (SerhiiKirik) [inline client/src/components/Filters/Inputs/SortDirectionField/SortDirectionField.tsx:8]: move to the first line

File: client/src/components/MovieCard/MovieCardAlert/MovieCardAlert.tsx [hunk 3]
```diff
@@ -16,6 +17,12 @@ export const MovieCardAlert: FC<Props> = (props) => {
     onChangeAlert,
   } = props;
 
+  const intl = useIntl();
+
+  const message = cardAction === CARD_ACTION.ActionAdded
+    ? intl.formatMessage({ id: 'movieCardAlert.added' })
+    : intl.formatMessage({ id: 'movieCardAlert.deleted' });
```
Comment (SerhiiKirik) [inline client/src/components/MovieCard/MovieCardAlert/MovieCardAlert.tsx:24]:
```suggestion
  const message = useMemo(() => intl.formatMessage({
     id: 'movieCardAlert.${cardAction}' 
   }}, []);
```

File: client/src/components/Navigation/SelectOfTranslation/SelectItem/SelectItem.tsx [hunk 4]
```diff
@@ -1,27 +1,35 @@
...
 
 export const SelectItem: FC<Props> = (props) => {
   const {
     state,
     language,
     onChangeLanguage,
+    handleClose,
   } = props;
 
+  const text = language === 'en-us' ? 'USA' : 'UKR';
```
Comment (SerhiiKirik) [inline client/src/components/Navigation/SelectOfTranslation/SelectItem/SelectItem.tsx:21]: memo?

File: client/src/components/Navigation/SelectOfTranslation/SelectItem/SelectItem.tsx [hunk 5]
```diff
@@ -1,27 +1,35 @@
...
+
   return (
     <MenuItem
       value={language}
       disabled={state.locale === language}
-      onClick={() => onChangeLanguage(language)}
+      onClick={() => {
```
Comment (SerhiiKirik) [inline client/src/components/Navigation/SelectOfTranslation/SelectItem/SelectItem.tsx:27]: make a separate function for this
```suggestion
      onClick={handleClick}
```

File: client/src/components/Navigation/SelectOfTranslation/SelectOfTranslation.tsx [hunk 6]
```diff
@@ -19,33 +20,46 @@ interface Props {
...
   const [speech, setSpeech] = useState('');
+  const [isOpen, setIsOpen] = useState(false);
 
-  const handleChange = useCallback((event: SelectChangeEvent) => {
+  const handleChange = (event: SelectChangeEvent) => {
     setSpeech(event.target.value);
-  }, []);
+  };
+
+  const handleChangeOpening = () => {
```
Comment (SerhiiKirik) [inline client/src/components/Navigation/SelectOfTranslation/SelectOfTranslation.tsx:36]: useCallback

File: client/src/context/appContext/appContext.tsx [hunk 7]
```diff
@@ -1,39 +1,50 @@
...
+import { useLanguageStorage } from '../../hooks/useLanguageStorage';
+import { LOCALES } from './const';
 
 interface AppContextProviderProps {
-  children: React.ReactNode;
+  children: ReactNode;
 }
 
-interface Action {
+export interface Action {
```
Comment (SerhiiKirik) [inline client/src/context/appContext/appContext.tsx:12]: move to typeDefs

File: client/src/context/i18n/i18n.tsx [hunk 8]
```diff
@@ -0,0 +1,25 @@
+import React, { FC, Fragment } from 'react';
+import { IntlProvider } from 'react-intl';
+import flatten from 'flat';
+import messages from '../../messages';
+
+import { LOCALES } from '../appContext/const';
+
+interface Props {
+  children: any,
```
Comment (SerhiiKirik) [inline client/src/context/i18n/i18n.tsx:9]:
```suggestion
  children: React.Children;
```
OR
```suggestion
  children: React.JSX..;
```

**Review by SerhiiKirik — CHANGES_REQUESTED** [review]
Great 🔥 
Keep it up 🚀

File: client/src/context/appContext/appContext.tsx [hunk 9]
```diff
@@ -1,39 +1,50 @@
...
   dispatch: () => null,
 });
 
 const reducer = (state: any, action: Action) => {
   switch (action.type) {
-    case 'reset':
-      return defaultContext;
     case 'setLocale':
+      // eslint-disable-next-line react-hooks/rules-of-hooks
+      useLanguageStorage('locale', action.locale);
```
[the remark this answers is not in the document]
Comment (VolodymyrKirichenko) [reply client/src/context/appContext/appContext.tsx:35]: Ohh, i know that hooks can only be used in functional components or in other hooks, and i think - i know how to fix it

File: client/src/components/AboutMovie/AboutMovieMedia/AboutMovieMedia.tsx [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+import { ModalWindow } from './ModalWindow/ModalWindow';
+
+interface Props {
+  movie: Movie,
+}
+
+export const AboutMovieMedia: FC<Props> = (props) => {
+  const { movie } = props;
+
+  const [open, setOpen] = useState(false);
```
Comment (SerhiiKirik) [inline client/src/components/AboutMovie/AboutMovieMedia/AboutMovieMedia.tsx:14]:
```suggestion
  const [isOpen, setIsOpen] = useState(false);
```

File: client/src/components/Navigation/SelectOfTranslation/SelectOfTranslation.tsx [hunk 11]
```diff
@@ -19,33 +20,52 @@ interface Props {
...
 
   const [speech, setSpeech] = useState('');
+  const [isOpen, setIsOpen] = useState(false);
 
-  const handleChange = useCallback((event: SelectChangeEvent) => {
+  const handleChange = (event: SelectChangeEvent) => {
     setSpeech(event.target.value);
+  };
+
+  const handleChangeOpening = useCallback(() => {
```
[the remark this answers is not in the document]
Comment (SerhiiKirik) [reply client/src/components/Navigation/SelectOfTranslation/SelectOfTranslation.tsx:38]: create a hook)

File: client/src/components/typedefs/typedefs.tsx [hunk 12]
```diff
@@ -48,6 +66,34 @@ export interface InitialValues {
...
+
+export interface Errors {
+  listName?: string;
+}
+
+export interface Values {
+  listName?: string;
+}
+
+export interface MoviesByIds {
```
Comment (SerhiiKirik) [inline client/src/components/typedefs/typedefs.tsx:93]: Here just movies, mb would be better give a name like `Movies` ?

File: client/src/context/i18n/i18n.tsx [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+export const I18Provider: FC<Props> = (props) => {
+  const { children, locale = LOCALES.ENGLISH } = props;
+
+  return (
+    <IntlProvider
+      textComponent={Fragment}
+      locale={locale}
+      messages={flatten(messages[locale])}
```
Comment (SerhiiKirik) [inline client/src/context/i18n/i18n.tsx:20]: move to the variable
```suggestion
      messages={message}
```

**Review by SerhiiKirik — APPROVED** [review]
Great 🔥

File: client/src/components/typedefs/typedefs.tsx (hunk 12 above)
Comment (VolodymyrKirichenko) [reply client/src/components/typedefs/typedefs.tsx:93]: I think no, because I already have one interface for just ```Movies```

**Pull request merged**